lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -252,3 +326,169 @@ impl OperationDurationsAggregator {
         data.iter().sum::<f32>() / data.len() as f32
     }
 }
+
+/// Convert a fixed-size non-cumulative histogram to a sparse cumulative histogram.
+/// Omit repeated values to reduce the size of the histogram.
+fn convert_histogram(
+    le_boundaries: &[f32],
+    counts: &[usize],
+    total_count: usize,
+) -> Vec<(f32, usize)> {
+    let mut result = Vec::new();
```

Comment:
Maybe we can reserve a default capacity like `Vec::with_capacity(le_boundaries.len())` here to prevent some reallocations once the list grows.